src/riot-rs-embassy/src/network.rs

```diff
@@ -14,12 +11,10 @@ pub const ETHERNET_MTU: usize = 1514;
 
 pub type NetworkStack = Stack<NetworkDevice>;
 
-pub(crate) static STACK: CriticalSectionMutex<OnceCell<SendCell<&'static NetworkStack>>> =
-    CriticalSectionMutex::new(OnceCell::new());
+pub(crate) static STACK: OnceLock<SendCell<&'static NetworkStack>> = OnceLock::new();
```

Comment:
I think we don't need the `SendCell` here anymore, because `OnceLock` implements `Sync`.

Reply:
That doesn't help. We're using the `SendCell` to prevent the stack from being accessed from other executors. The `Sync` is necessary here to access the global static in the first place. But IIUC `Sync` on `OnceLock` doesn't make it safe to send `NetworkStack` to another thread / executor.

Reply:
I think there are actually some trait bounds missing on `OnceLock`'s `Sync` impl.

Reply:
Hmm, maybe you're right. But in that case I think it's problematic that `OnceLock` blindly implements `Sync`, because it would have allowed our code to compile without the `SendCell`.